libs/detectors/x86/detector.py

```diff
@@ -7,18 +41,23 @@ class Detector:
     input image in order to get the detection results.
 
     :param config: Is a ConfigEngine instance which provides necessary parameters.
+    :param source: A string that represents the camera the camera. Ex: "Source_1".
     """
 
-    def __init__(self, config):
+    def __init__(self, config, source):
         self.config = config
-        self.name = self.config.get_section_dict('Detector')['Name']
+        camera_id = self.config.get_section_dict(source)["Id"]
+        device = self.config.get_section_dict("Detector")["Device"]
+        model_data = get_or_create_model_json_file(self.config, device, camera_id)
+
+        self.name = model_data["model_name"]
 
         if self.name == 'mobilenet_ssd_v2':
             from libs.detectors.x86 import mobilenet_ssd
             self.net = mobilenet_ssd.Detector(self.config)
         elif self.name == "openvino":
             from libs.detectors.x86 import openvino
-            self.net = openvino.Detector(self.config)
+            self.net = openvino.Detector(self.config, self.name, model_data["variables"])
```

Comment:
This change must be made for all detectors and their models.

Reply:
Agree

Reply:
Of course, this commit was just a proof of concept in order to validate these changes. I will generalize it.